extensions/wikia/SpecialDiscussionsLog/SpecialDiscussionsLogController.class.php

```diff
@@ -297,4 +297,25 @@ private function checkAccess() {
 			throw new \PermissionsError( self::DISCUSSIONS_LOG_ACTION );
 		}
 	}
+
+	private function getLocationFromIP( $ip, array &$ipCache ) {
+		if (array_key_exists($ip, $ipCache)) {
+			return $ipCache[$ip];
+		}
+
+		$details = json_decode(file_get_contents("http://ipinfo.io/{$ip}/json"));
+		$location_arr = array($details->country);
+
+		if ( $details->region ) {
+			array_unshift($location_arr, $details->region);
+		}
+
+		if ( $details->city ) {
+			array_unshift($location_arr, $details->city);
+		}
```

Comment:
So it looks like you're using `array_unshift` to get these in a City, Region, Country order.  However, unshifting wouldn't be necessary if you added these in the order you want to begin with.  Then you could use the standard PHP array push syntax, e.g.:
```php
$location_arr[] = $details->city
```